webroot/adm/client-key.html

```diff
@@ -105,69 +102,105 @@ <h3>Output</h3>
                 return;
             }
 
-            var clientName = encodeURIComponent($('#clientName').val());
-            var url = '/api/client/del?name=' + clientName;
+            var clientContact = encodeURIComponent($('#clientContact').val());
+            var url = '/api/client/del?contact=' + clientContact;
 
             doApiCall('POST', url, '#standardOutput', '#errorOutput');
         });
 
         $('#doSetRoles').on('click', function () {
-            var clientName = encodeURIComponent($('#clientName').val());
-            var roles = encodeURIComponent($('#roles').val());
-            var url = '/api/client/roles?name=' + clientName + '&roles=' + roles;
+            var clientContact = encodeURIComponent($('#clientContact').val());
+            var roles = prompt("Enter the new roles", "");
+            if (!roles){
+                if (roles === '') $('#errorOutput').html("Required Parameter: new roles");
+                return;
+            }
+
+            var url = '/api/client/roles?contact=' + clientContact + '&roles=' + roles;
 
             doApiCall('POST', url, '#standardOutput', '#errorOutput');
         });
 
         $('#doSetSite').on('click', function () {
-            var clientName = encodeURIComponent($('#clientName').val());
-            var siteId = encodeURIComponent($('#siteId').val());
-            var serviceId = encodeURIComponent($('#serviceId').val());
-            var url = '/api/client/update?name=' + clientName + '&site_id=' + siteId + '&service_id=' + serviceId;
+            var clientContact = encodeURIComponent($('#clientContact').val());
+
+            var siteId = prompt("Enter the new siteId", "");
+            if (!siteId){
+                if (siteId === '') $('#errorOutput').html("Required Parameter: siteId");
+                return;
+            }
+            var serviceId = prompt("Enter the new serviceId", "");
```

Comment:
Do you know if there is some link between `serviceId` and `siteId`? To me, they are distinct, and probably should be separate buttons.

Reply:
The update site was utilizing this endpoint already which updated both. I believe that serviceId is related to the siteId in some manner however so should be changed.